src/Content/MemoryContentStorageHandler.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cache\MemoryCache;
+use Kirby\Cms\Language;
+use Kirby\Cms\ModelWithContent;
+
+/**
+ * @package   Kirby Content
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class MemoryContentStorageHandler extends ContentStorageHandler
+{
+	/**
+	 * Cache instance, used to store content in memory
+	 */
+	protected MemoryCache $cache;
+
+	/**
+	 * Sets up the cache instance
+	 */
+	public function __construct(protected ModelWithContent $model)
+	{
+		parent::__construct($model);
+		$this->cache = new MemoryCache();
+	}
+
+	/**
+	 * Returns a unique id for a combination
+	 * of the version id, the language code and the model id
+	 */
+	protected function cacheId(VersionId $versionId, Language $language): string
+	{
+		return $versionId->value() . '/' . $language->code() . '/' . $this->model->id();
+	}
+
+	/**
+	 * Deletes an existing version in an idempotent way if it was already deleted
+	 */
+	public function delete(VersionId $versionId, Language $language): void
+	{
+		$this->cache->remove($this->cacheId($versionId, $language));
+	}
+
+	/**
+	 * Checks if a version exists
+	 */
+	public function exists(VersionId $versionId, Language $language): bool
+	{
+		$data = $this->cache->get($this->cacheId($versionId, $language));
+
+		// validate the cached data
+		if (
+			isset($data['fields']) === true &&
+			is_array($data['fields']) === true &&
+			isset($data['modified']) === true &&
+			is_int($data['modified']) === true
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the modification timestamp of a version if it exists
+	 */
+	public function modified(VersionId $versionId, Language $language): int|null
+	{
+		if ($this->exists($versionId, $language) === false) {
+			return null;
+		}
+
+		return $this->cache->get($this->cacheId($versionId, $language))['modified'];
+	}
+
+	/**
+	 * Returns the stored content fields
+	 *
+	 * @return array<string, string>
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
+	 */
+	public function read(VersionId $versionId, Language $language): array
+	{
+		$this->ensure($versionId, $language);
+		return $this->cache->get($this->cacheId($versionId, $language))['fields'];
+	}
+
+	/**
+	 * Updates the modification timestamp of an existing version
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
+	 */
+	public function touch(VersionId $versionId, Language $language): void
+	{
+		$fields = $this->read($versionId, $language);
+		$this->write($versionId, $language, $fields);
```

Comment:
`Suggestion`: For this we could add `$cache->touch()` that updates the created/modified time of the cache `Value`. I think that could be universally useful.

Reply:
I like the idea, but I shy away from implementing it for all cache handlers in this PR.

Reply:
You are right, I think it's still fine the way you've done it.